### PR TITLE
Update docs & extend docs for test setup

### LINKED ISSUE #368: Add documentation about the test setup
We should add documentation about the following points into MkDocs:

- [ ] How to run and extend the integration tests.
- [ ] What helper functionalities are provided and how to use them.
- [ ] Revise the existing documentation for the C++ tests.

### PR DESCRIPTION
**Pull request by mmeijerdfki**
Closes #368.

Changes to highlight for discussion:

- removal of code in the docs, instead linking the code snippets using [pymdown-extensions](https://facelessuser.github.io/pymdown-extensions/extensions/snippets/) in conjunction with raw.githubusercontent.com.
- Addition of the `.drawio` file of the diagram to explain the message abstractions into the source control.

The first point brings the pro that updates are directly reflected (when the docs are updated through mkdocs), but imposes the problem that snippets are specified using line numbers and those possibly would've to be updated as well.

### CODE REVIEW

File: docs/mkdocs/reference/pytests-message-abstractions.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+`_set_enum_func_mapping()`, which is discussed in the previous section.
+
+Lastly if a change to the dictionary has been done using `set_active_function()` or `set_mapped_functions()`, the `assemble_datatype_instance()`
+has to be called to reassemble the underlying datatype.
+
+**Note**: On creation of the instance the datatype is assembled automatically in it's constructor.
+
+Now the assembled datatype can be accessed using the property `queryinst_builder.datatype_instance` and
+the underlying flatbuffers builder can be accessed by using `queryinst_builder.builder` for further use,
+e.g. like in this case calling the service function `call_get_instances_fb()` using the [ServiceManager](http://127.0.0.1:8001/reference/python-helpers/#service_managerpy).
```
Comment (Mark-Niemeyer) [inline docs/mkdocs/reference/pytests-message-abstractions.md:88]: Make this a relative path. This is the path from your local test deployment.

**Review by Mark-Niemeyer — CHANGES_REQUESTED** [review]

**Comment by Mark-Niemeyer** [comment]
> but imposes the problem that snippets are specified using line numbers and those possibly would've to be updated as well.

We could add comments to the python files so that the person who wants to change them is informed that it effects the docs

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -27,6 +27,7 @@ repos:
       - id: check-xml
       - id: check-yaml
         args: [--allow-multiple-documents]
+        exclude: mkdocs.yml
```
[the remark this answers is not in the document]
Comment (mmeijerdfki) [reply .pre-commit-config.yaml:30]: The checker doesn't allow for custom tags as discussed in the issue at https://github.com/pre-commit/pre-commit-hooks/issues/103. 
The same problem as in [this issue](https://github.com/pre-commit/pre-commit-hooks/issues/567) occurs.
For the extensions we use the same custom tags functionality:
https://github.com/agri-gaia/seerep/blob/7dddbf7af2b0336509e8efad95f03ed5e01b802a/mkdocs.yml#L59-L61

Now another workaround is implemented, which uses the `--unsafe` option just for the `mkdocs.yml`.

**Comment by mmeijerdfki** [comment]
I added a notice when files are referenced in the `mkdocs`. I also updated some links in the docs, such that `mkdocs` is happier and doesn't print as many messages as before.

**Review by Mark-Niemeyer — APPROVED** [review]

**Pull request merged**